server/resource/caldav_functions.go

```diff
@@ -0,0 +1,133 @@
+package resource
+
+import (
+	"errors"
+	"github.com/daptin/daptin/server/statementbuilder"
+	"github.com/doug-martin/goqu/v9"
+	"time"
+)
+
+func(dr *DbResource) GetRpath(userId int64)(string,error){
+
+	rPath := ""
+
+	cal, _, err := dr.Cruds["calendar"].GetRowsByWhereClause("calendar", nil, goqu.Ex{"owner_id": userId})
+	if err != nil {
+		return rPath, err
+	}
+	if len(cal) < 1 {
+		return rPath, errors.New("calendar not found")
+	}
+
+	rPath = cal[0]["rPath"].(string)
+
+	return rPath, err
+}
+
+func (dr *DbResource) DeleteCalendarEvent(UserId int64, rPath string) error {
+
+	cal, err := dr.Cruds["calendar"].GetAllObjectsWithWhere("calendar",
+		goqu.Ex{
+			"owner_id": UserId,
+			"rPath": rPath,
+		},
+	)
+
+	if err != nil || len(cal) == 0 {
+		return errors.New("caldav resource does not exist")
+	}
+
+	query, args, err := statementbuilder.Squirrel.Delete("calendar").Where(goqu.Ex{"rPath": cal[0]["rPath"]}).ToSQL()
+	if err != nil {
+		return err
+	}
+
+	_, err = dr.db.Exec(query, args...)
+	if err != nil {
+		return err
+	}
+
+
+	return err
+
+}
+
+func(dr *DbResource) GetModTime(rPath string, userId int64)(time.Time,error){
+	modified := time.Now()
+
+	cal, _, err := dr.Cruds["calendar"].GetRowsByWhereClause("calendar", nil, goqu.Ex{"owner_id": userId},goqu.Ex{"rPath": rPath})
+	if err != nil {
+		return modified, err
+	}
+
+	if len(cal) < 1 {
+		return modified, errors.New("calendar not found")
+	}
+
+	modified = cal[0]["last_modified"].(time.Time)
+
+	return modified, err
+}
+
+func(dr *DbResource) GetContent(rPath string, userId int64)(string,error){
+	content := ""
+
+	cal, _, err := dr.Cruds["calendar"].GetRowsByWhereClause("calendar", nil, goqu.Ex{"owner_id": userId},goqu.Ex{"rPath": rPath})
```

Comment:
Use `user_account_id` (Foreign key to user_account.id) column instead of `owner_id` for row ownership